o1vm/src/pickles/tests.rs

```diff
@@ -56,3 +68,46 @@ fn test_regression_selectors_for_instructions() {
         .iter()
         .for_each(|c| assert!(c.degree(1, 0) == 2 || c.degree(1, 0) == 1));
 }
+
+fn zero_to_n_minus_one(n: usize) -> Vec<Fq> {
+    (0..n).map(|i| Fq::from((i) as u64)).collect()
+}
+#[test]
+fn test_small_circuit() {
+    let domain = EvaluationDomains::<Fq>::create(8).unwrap();
+    let srs = SRS::create(8);
+    let proof_input = ProofInputs::<Pallas> {
+        evaluations: WitnessColumns {
+            scratch: std::array::from_fn(|_| zero_to_n_minus_one(8)),
+            instruction_counter: zero_to_n_minus_one(8)
+                .into_iter()
+                .map(|x| x + Fq::one())
+                .collect(),
+            error: (0..8)
+                .map(|i| -Fq::from((i * SCRATCH_SIZE + (i + 1)) as u64))
+                .collect(),
+            selector: zero_to_n_minus_one(8),
+        },
+    };
+    let mut expr = Expr::literal(Fq::zero());
+    for i in 0..SCRATCH_SIZE + 2 {
+        expr += Expr::cell(Column::Relation(i), CurrOrNext::Curr);
+    }
+    /* expr *= Expr::cell(Column::DynamicSelector(0), CurrOrNext::Curr); */
+    let mut rng = make_test_rng(None);
+    type BaseSponge = DefaultFqSponge<PallasParameters, PlonkSpongeConstantsKimchi>;
```

Comment:
Can we move this in `mod.rs`?

Reply:
I would disagree.  These are only used in the tests, and so belong here.

Reply:
It is not used only in tests. There are also the same definitions in `mod.rs` for `main.rs`.